src/ui/components/BalanceInput.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {useEffect, useState} from 'react';
+import {StyleSheet} from 'react-native';
+import {TextInput, useTheme} from '@ui/library';
+import {useFormatBalance} from 'src/api/hooks/useFormatBalance';
+import {decimalKeypad} from 'src/utils';
+import MaxBalance from './MaxBalance';
+import {getBalanceFromString} from 'src/api/utils/balance';
+import {ApiPromise} from '@polkadot/api';
+import type {Account} from 'src/api/hooks/useAccount';
+
+type PropTypes = {
+  api?: ApiPromise;
+  account?: Account;
+  onChangeBalance: (dispatch: string) => void;
+};
+
+export function BalanceInput(props: PropTypes) {
+  const {colors} = useTheme();
+  const {api, account} = props;
+  const [amount, setAmount] = useState<string>('');
+  const [isMaximum, setIsMaximum] = useState<boolean>(true);
+  const formatBalance = useFormatBalance();
+  useEffect(() => {
+    amount !== '' && Number(amount) >= Number(account?.balance.total) ? setIsMaximum(false) : setIsMaximum(true);
+  }, [amount, account]);
+  return (
+    <>
+      <TextInput
+        dense
+        style={styles.textInput}
+        theme={{colors: {primary: isMaximum ? colors.primary : colors.backdrop}}}
```

Comment:
```suggestion
        error={isMaximum}
```

Also the logic `amount !== '' && Number(amount) >= Number(account?.balance.total) ? setIsMaximum(false) : setIsMaximum(true);` is confusing. You are setting `isMaximum` to false if the entered amount is higher than the balance. Maybe reverse it?